kernel/Kconfig

```diff
@@ -910,6 +910,15 @@ config KERNEL_COHERENCE
 	  may fail strangely.  Some assertions exist to catch these
 	  mistakes, but not all circumstances can be tested.
 
+config KERNEL_COHERENCE_PIPES
+	bool "Safely use pipes on a cache-incoherent architecture"
+	default y if KERNEL_COHERENCE
+	help
+	  When enabled, this option allows pipes to be safely used on a
+	  cache-incoherent architecture. If pipes are not being used on such
+	  a system, it is recommended that this option be disabled so that
+	  some small amount of space in 'struct k_thread' can be saved.
```

Comment:
Not loving this as an API choice.  Clearly the intent is that you want an app author to be able to say "I don't care about pipes" and save a word.  But having made that choice, nothing tells future authors about it.  The effect is that you end up with a configuration where the pipe API works, mostly, except across CPUs in subtle and weirdly broken ways. 

I think if you want to make this optional, it's probably better to just have "config KERNEL_PIPE" to en/disable the whole API rather than have this thing that ties two otherwise unrelated kconfigs together.

Reply:
It is easy enough to alter things to make pipes a configurable element and have them always use the 'pipe descriptor' stored in the thread. I have no objections to doing that.